ckan/lib/helpers.py

```diff
@@ -50,18 +50,37 @@
 log = logging.getLogger(__name__)
 
 
+class HelperError(Exception):
+    """Raised if an attempt to access an undefined helper is made.
+
+    Normally, this would be a subclass of AttributeError, but Jinja2 will
+    catch and ignore them. We want this to be an explicit failure re #2908.
+    """
+
+
 class AttributeDict(dict):
```

Comment:
name this HelperDict? We have a custom Helper-specific error now (and I'd rather not encourage the use of AttributeDict in other parts of the code... python is not JS!)